packages/lib-user/src/components/Contributors/components/ExportStats/ExportStats.js

```diff
@@ -0,0 +1,216 @@
+import { Box, Button, Layer, Text, Anchor } from 'grommet'
+import { Alert, Download } from 'grommet-icons'
+import { func, number, string, shape } from 'prop-types'
+
+import { useTranslation } from '../../../../translations/i18n.js'
+
+function ExportBox({ children }) {
+  return (
+    <Box
+      align='center'
+      gap='small'
+      height='medium'
+      justify='center'
+      pad='medium'
+      width={{ min: 'medium' }}
+    >
+      {children}
+    </Box>
+  )
+}
+
+const DEFAULT_DOWNLOAD_URL = {
+  url: '',
+  filename: ''
+}
+const DEFAULT_HANDLER = () => true
+
+function ExportStats({
+  csvSizeEstimate = '',
+  downloadUrl = DEFAULT_DOWNLOAD_URL,
+  errorMessage = '',
+  exportProgress = 0,
+  exportStatus = asyncStates.initialized,
+  memberCount = 0,
+  onClose = DEFAULT_HANDLER,
+  onConfirm = DEFAULT_HANDLER,
+  onRetry = DEFAULT_HANDLER
+}) {
+  const { t } = useTranslation()
+  const loadingExportMessage = t('Contributors.ExportStats.downloading')
+
+  // loading state
+  if (exportStatus === asyncStates.loading) {
+    return (
+      <Layer>
+        <ExportBox>
+          <Text>
+            {loadingExportMessage}
+          </Text>
+          <Loader loadingMessage={loadingExportMessage} />
+          <Text>
+            {t('Contributors.ExportStats.progress', {
+              progress: Math.round(exportProgress)
+            })}
+          </Text>
+        </ExportBox>
+      </Layer>
+    )
+  }
+
+  // error state
+  if (exportStatus === asyncStates.error) {
+    return (
+      <Layer>
+        <ExportBox>
+          <Box
+            align='center'
+            gap='small'
+          >
+            <Alert
+              color='status-error'
+              size='large'
+            />
+            <Text
+              textAlign='center'
+              weight='bold'
+              color='status-error'
+            >
+              {t('Contributors.ExportStats.error')}
+            </Text>
+            <Text textAlign='center'>
+              {errorMessage || t('Contributors.ExportStats.errorMessage')}
+            </Text>
+          </Box>
+          <Box
+            direction='row'
+            gap='medium'
+            margin={{ top: 'medium' }}
+          >
+            <Button 
+              label={t('Contributors.ExportStats.close')}
+              onClick={onClose}
+            />
+            <Button 
+              label={t('Contributors.ExportStats.retry')}
+              onClick={onRetry}
+              primary
+            />
+          </Box>
+        </ExportBox>
+      </Layer>
+    )
+  }
+
+  // success state without filename
+  if (exportStatus === asyncStates.success && !downloadUrl.filename) {
+    return (
+      <Layer>
+        <ExportBox>
+          <Text
+            textAlign='center'
+          >
+            {t('Contributors.ExportStats.generating')}
+          </Text>
+          <Button
+            label={t('Contributors.ExportStats.close')}
+            onClick={onClose}
```

Comment:
Since the browser might be hanging at this stage while the CSV content is created, maybe the Close button should be `disabled`? I lean towards the button _not_ be disabled, but open to suggestion.
```suggestion
            disabled
            onClick={onClose}
```

Reply:
Will the `onClose` function cancel the csv generation?

I don't think the button should be disabled, but the `Contributors.ExportStats.generating` could offer guidance such as "Generating stats export file. Do not close this browser tab".